code/modules/admin/IsBanned.dm

```diff
@@ -126,10 +126,32 @@
 
 		var/newmatch = FALSE
 		var/client/C = GLOB.directory[ckey]
-		var/cachedban = SSstickyban.cache[bannedckey]
+		var/list/cachedban = SSstickyban.cache[bannedckey]
+		if (!CONFIG_GET(flag/ban_legacy_system) && (SSdbcore.Connect() || length(SSstickyban.dbcache)))
+			ban = get_stickyban_from_ckey(bannedckey)
+			var/list/bancache = list()
+			while (ban["ckey"] && ban["keys"] && ban["keys"][ckey] && ban["keys"][ckey]["exempt"])
+				if (C || SSstickyban.confirmed_exempt[ckey]) //modifing/re-adding a stickyban makes it re-match everybody
+					return
+				bancache[ban["ckey"]] = world.GetConfig("ban", ban["ckey"])
+				//Hacky way to ensure somebody exempt from one stickyban doesn't get exempt from all stickybans
+				world.SetConfig("ban", bannedckey, null)

```

Comment:
how does this work?

Reply:
We remove the current ban, then re-call parent to see if it matches any other stickybans.

At the end we re-add the ban(s) that were removed.

We cache the checks because byond will re-call isbanned() when we re-add the stickybans. I need to add some checks against that happening so the rest of isbanned() doesn't get called.